### ISSUE TITLE
Move cosine LR scheduler with warmup from modules/ to training/

### ISSUE THREAD

**Issue by ebsmothers** [pytorch/torchtune#1587]
Hopefully not a controversial suggestion? I think training is the more natural place for LR schedulers (unless there is some dependency issue I'm missing).

**Comment by krammnic** [comment]
It seem to me that this affects on recipes, docs and 1 test. Will open PR now.

**Comment by ebsmothers** [comment]
Thanks for creating the PR! I agree, let's keep the one in modules for now (but point everything to the version in training). Then add the deprecated flag saying that it will be removed in the 0.4 release and to use the one in training/ instead ([example usage](https://github.com/pytorch/torchtune/blob/5de500147e57d4d35e296faba68b296c03a1a2a0/torchtune/models/gemma/transformer.py#L16))

**Comment by joecummings** [comment]
Closed by #1801
